src/components/BuildBaronAndAnnotations/AnnotationNote.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  EditAnnotationNoteMutation,
+  EditAnnotationNoteMutationVariables,
+  Note,
+} from "gql/generated/types";
+import { useBannerDispatchContext } from "../../context/banners";
+import { EDIT_ANNOTATION_NOTE } from "../../gql/mutations/edit-annotation-note";
+import { getDateCopy } from "../../utils/string";
+import { ConditionalWrapper } from "../ConditionalWrapper";
+import { TicketsTitle, TitleAndButtons, MetaDataWrapper } from "./BBComponents";
+
+interface Props {
+  note: Note;
+  taskId: string;
+  execution: number;
+  userCanModify: boolean;
+}
+
+export const AnnotationNote: React.FC<Props> = ({
+  note,
+  taskId,
+  execution,
+  userCanModify,
+}) => {
+  const originalMessage = note?.message;
+  const dispatchBanner = useBannerDispatchContext();
+  const [newMessage, setMessage] = useState(originalMessage);
+  const [
+    updateAnnotationNote,
+    { loading: loadingAnnotationNote },
+  ] = useMutation<
+    EditAnnotationNoteMutation,
+    EditAnnotationNoteMutationVariables
+  >(EDIT_ANNOTATION_NOTE, {
+    onCompleted: () => {
+      dispatchBanner.successBanner(`Annotation note updated.`);
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error updating this note: ${error.message}`
+      );
+    },
+    refetchQueries: ["GetTask"],
+  });
+  const saveAnnotationNote = () => {
+    updateAnnotationNote({
+      variables: {
+        taskId,
+        execution,
+        originalMessage,
+        newMessage,
+      },
+    });
+  };
+
+  return (
+    <TitleAndButtons>
+      <TicketsTitle>Note</TicketsTitle>
+      {note && (
+        <>
+          <MetaDataWrapper data-cy={`${originalMessage}-metadata`}>
+            <Disclaimer>
+              Updated: {getDateCopy(note.source.time, null, true)}{" "}
+            </Disclaimer>
+            <Disclaimer>Last Edited By: {note.source.author}</Disclaimer>
+          </MetaDataWrapper>
+        </>
+      )}
+      <TextArea
+        id="noteInput"
+        rows={2}
+        style={{ width: 600 }}
+        value={newMessage}
+        onChange={(e) => setMessage(e.target.value)}
+      />
+      <ConditionalWrapper
+        condition={!userCanModify}
+        wrapper={(children) => (
+          <Tooltip title="You are not authorized to edit task annotations">
+            <span>{children}</span>
+          </Tooltip>
+        )}
+      >
+        <ButtonWrapper>
+          <Button
+            data-cy="edit-annotation-button"
+            variant={Variant.Primary}
+            size="xsmall"
+            loading={loadingAnnotationNote}
+            onClick={saveAnnotationNote}
+            disabled={originalMessage === newMessage || !userCanModify}
+          >
+            Save Note
+          </Button>
+        </ButtonWrapper>
+      </ConditionalWrapper>
+    </TitleAndButtons>
+  );
+};
+
+export const ButtonWrapper = styled.div`
+  margin-right: 8px;
+  padding-top: 5px;
+`;
+const { TextArea } = Input;
```

Comment:
We usually do this at the top under the imports like 
https://github.com/evergreen-ci/spruce/blob/7f35ea9ee8d6230fee42a492ba52a2d7d2302733/src/components/BuildBaronAndAnnotations/AddIssueModal.tsx#L16

Reply:
Gotcha, thanks! Separately, I got the information about how to define TextArea from the internet, but I'm confused about how it works like this. I would think that I could add "rows" to input and have it translate to area, or that I would import TextArea instead of importing Input, but that's not the case. It seems instead like TextArea is a subclass of Input? Can you explain how this works? (or help me figure out what to google haha)

Reply:
Your exactly right antd has TextArea declared as a subclass within `Input`
![image](https://user-images.githubusercontent.com/4605522/103947463-16da7000-5106-11eb-8680-9d489c1fc256.png)

Reply:
You can cmd+click on almost any variable or imported variable in vscode and jump to its definition if you're curious.

Reply:
Thank you!